tests/JsonSchema/TypeFactoryTest.php

```diff
@@ -35,13 +35,89 @@ public function testGetType(array $schema, Type $type): void
     public function typeProvider(): iterable
     {
         yield [['type' => 'integer'], new Type(Type::BUILTIN_TYPE_INT)];
+        yield [['type' => ['integer', 'null']], new Type(Type::BUILTIN_TYPE_INT, true)];
         yield [['type' => 'number'], new Type(Type::BUILTIN_TYPE_FLOAT)];
+        yield [['type' => ['number', 'null']], new Type(Type::BUILTIN_TYPE_FLOAT, true)];
         yield [['type' => 'boolean'], new Type(Type::BUILTIN_TYPE_BOOL)];
-        yield [['type' => 'string'], new Type(Type::BUILTIN_TYPE_OBJECT)];
+        yield [['type' => ['boolean', 'null']], new Type(Type::BUILTIN_TYPE_BOOL, true)];
+        yield [['type' => 'string'], new Type(Type::BUILTIN_TYPE_STRING)];
+        yield [['type' => ['string', 'null']], new Type(Type::BUILTIN_TYPE_STRING, true)];
+        yield [['type' => 'object'], new Type(Type::BUILTIN_TYPE_OBJECT)];
```

Comment:
Please mind this change: I don't understand why `object` was being mapped as `string` previously, but I suppose the idea was to map UUID-alike single-field value types? That is indeed wrong, unless I'm missing a basic use-case.

Reply:
Not it is a limitation: until very recently, API Platform where only able to deal with object mapped with the @ApiResource annotation. And because it’s an hypermedia framework, when such object are encountered, they are replaced by their IRI during the serialization. (An IRI is just a string with a specific format in JSON Schema).

Reply:
So if I pass in an `stdClass` it would be replaced by a dummy string at runtime also now?

Reply:
No, it should work now (since the last minor version), this code path has probably been forgotten. To be tested to be sure :)

Reply:
So is the diff correct, or does it need adjustments here?

Mostly trying to avoid an unintentional BC break, if this is supposed to stay as `"string"` :+1:

Reply:
It looks good to me

Reply:
Reverted: deferred to https://github.com/api-platform/core/issues/3403